tests/zfs-tests/tests/functional/cache/cache_010_pos.ksh

```diff
@@ -50,51 +50,55 @@ function cleanup_testenv
 	if [[ -n $lofidev ]]; then
 		if is_linux; then
 			losetup -d $lofidev
+		elif is_freebsd; then
+			mdconfig -du ${lofidev#md}
 		else
 			lofiadm -d $lofidev
 		fi
 	fi
 }
 
-log_assert "Cache device can only be block devices."
+log_assert "Verify cache devices can be disk, file, lofi device or any " \
+  "device that presents a block interface"
+
+verify_disk_count "$DISKS" 2
 log_onexit cleanup_testenv
 
 TESTVOL=testvol1$$
 dsk1=${DISKS%% *}
 log_must zpool create $TESTPOOL ${DISKS#$dsk1}
 
-# Add nomal ${DEV_RDSKDIR} device
+# Add normal ${DEV_RDSKDIR} device
 log_must zpool add $TESTPOOL cache \
-    ${DEV_RDSKDIR}/${dsk1}${SLICE_PREFIX}${SLICE0}
+    ${DEV_RDSKDIR}/${dsk1}
+log_must zpool remove $TESTPOOL ${DEV_RDSKDIR}/${dsk1}
+
+
+# Add provided disk
+log_must zpool add $TESTPOOL cache $dsk1
 log_must verify_cache_device $TESTPOOL $dsk1 'ONLINE'
+log_must zpool remove $TESTPOOL $dsk1
 
 # Add normal file
-log_mustnot zpool add $TESTPOOL cache $VDEV2
+log_must zpool add $TESTPOOL cache $LDEV
+ldev=$(random_get $LDEV)
```

Comment:
Unlike the `slog` tests it looks like `LDEV` isn't defined to be a file and is instead `"$DISK0`.  You could use `VDEV` instead here to use a file for the cache device.

I also noticed that `./functional/cache/setup.ksh` includes a related check which should be removed.

```sh
if ! is_physical_device $LDEV; then
        log_unsupported "Only physical disk could be cache device"
fi
```

Reply:
Done! Missed that one.